pkg/util/label/label.go

```diff
@@ -0,0 +1,44 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package label
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// The AdditionalLabels is a big string consisting of key=value, set at build time
+// when set are to be added to Deployments, CronJob and KNativeService whose pods
+// should expose these labels
+
+// AdditionalLabels are labels=values, they MUST be set as key=value separated by comma ,
+// example: myKey1=myValue1,myKey2=myValue2
+// Also it supports replacing a value for the integration name, just put @ around it
+// example: myKey1=myValue1,myKey2=@integration@
+var AdditionalLabels = ""
+
+// parses the AdditionalLabels variable and returns as map[string]string
+func AddLabels(integration string) map[string]string {
+	AdditionalLabels = strings.ReplaceAll(AdditionalLabels, "@integration@", integration)
+	AdditionalLabels = fmt.Sprintf("%s=%s,%s", v1.IntegrationLabel, integration, AdditionalLabels)
+	lbls, _ := labels.ConvertSelectorToLabelsMap(AdditionalLabels)
```

Comment:
Instead of swallowing the error, the parsing could be call in an init method, that would panic in case the labels are ill -defined.

Reply:
I think I've just experienced why fail-fast error handing is generally preferred. I tried to understand why the tests were failing, and it took me half an hour, scratching my head, to realise that there was a trailing comma in the string, among the zillion log statements printed in the e2e test suite execution, like the needle in a haystack. We are lucky Kubernetes caught the error, but that's the kind of things that could be hard to detect, and panicking would make it immediate.

Reply:
Let me precise my though, this is a variable set at build time, so the closer we fail to the build the better. Panicking at the operator start time is the closest place I can think of. It's correct it's not critical for the integrations, but it's rather _critical_ for the build. As with any feedback loop, the closer the the consequence from the cause the better.

Reply:
ok, I will panic then. Can you share which e2e failed, so I can reproduce ?

Reply:
One occurrence is available at: https://github.com/apache/camel-k/runs/3661980872?check_suite_focus=true#step:10:9045.

Reply:
I hadn't run the e2e tests before, good you had catch this one. just pushed the changes and hope it may be fixed now. Thanks.